packages/server/api/webscrape.py

```diff
@@ -3,32 +3,63 @@
 import re
 from urllib import parse
 from bs4 import BeautifulSoup
+import json
 
 class handler(BaseHTTPRequestHandler):
   whitelisted_url_regexes = [
     re.compile('^https:\/\/www\.bonappetit\.com')
   ]
 
+  def error_handler(error_string, error_code = 500):
+    self.send_response(error_code)
+    self.send_header('Content-type', 'text/plain')
+    self.end_headers()
+    self.wfile.write(errror_string.encode())
+
   def do_GET(self):
     url = parse.parse_qs(parse.urlsplit(self.path).query)['url'][0]
     url_in_whitelist = any(regex.match(url) for regex in self.whitelisted_url_regexes)
     
     if(not url_in_whitelist):
-      self.send_response(500)
-      self.send_header('Content-type', 'text/plain')
-      self.end_headers()
-      self.wfile.write('Unsupported recipe site'.encode())
+      error_handler('URL not supported')
```

Comment:
not calling with `self`

Reply:
Super wack, apparently you can't do that in python. Like the only way we could get it working is if we didn't pass self in. @ColinLMacLeod1 was very upset

Reply:
no no no, you have to call `self. error_handler`

Reply:
cuz the function is declared on the classs, right?

Reply:
https://www.makeuseof.com/tag/python-instance-static-class-methods/